### ISSUE TITLE
Falls back if the current version is newer than the latest

### ISSUE THREAD

**Issue by KEINOS** [nao1215/gup#36]
As shown below, the command `go-carpet` falls back from `v1.9.1` --> `v1.9.0`.

```
$ gup update
** snip **
gup:INFO : [18/25] github.com/msoap/go-carpet (v1.9.1-0.20220908165354-f7355b5d2afa to v1.9.0)
** snip **
```

Perhaps this is due to the way the version is defined during `go install`.

Since I wanted the "patched-latest-master" to be installed rather than the released version, I `go install`ed with the version tag as a branch name. Such as:

```go
go install "github.com/msoap/go-carpet@master"
```

The released version of this `go-carpet` command is v1.9.0. Though, as of Go v[1.18](https://go.dev/blog/go1.18) the `buildinfo` detects the version from the commit, it automatically increments the patch version. In this case from `v1.9.0` to `v1.9.1-0.20220908165354-f7355b5d2afa`.

But, the `gup update` will fall back if the version doesn't match to the latest.

https://github.com/nao1215/gup/blob/1e988d57f86f14c714e0703da75245d741f962fc/internal/goutil/goutil.go#L81-L84

### To Reproduce

```shellsession
$ gup check go-carpet
gup:WARN : not found 'go-carpet' package in $GOPATH/bin or $GOBIN
gup:FATAL: unable to check package: no package information

$ go install "github.com/msoap/go-carpet@master"

$ gup check go-carpet
gup:INFO : check binary under $GOPATH/bin or $GOBIN
gup:INFO : [1/1] github.com/msoap/go-carpet (current: v1.9.1-0.20220908165354-f7355b5d2afa, latest: v1.9.0)

gup:INFO : If you want to update binaries, run the following command.
           $ gup update go-carpet

$ gup update
** snip **
gup:INFO : [23/25] github.com/msoap/go-carpet (v1.9.1-0.20220908165354-f7355b5d2afa to v1.9.0)
** snip **
```

### Expected behavior

If the current version is greater than the latest, then do not fall back. Or at least ask the user to fall back.

### Env info

- OS: macOS Monterey (OSX 12.5)
- `gup`: v0.10.5
- `go`: v1.19.1 (Darwin/amd64)

**Comment by KEINOS** [comment]
## Proposed fix

Even though, it is ideal to use [go-semver](https://github.com/coreos/go-semver) for comparing the versions, I think the easiest fix is to use `strings.Compare()`.

```diff
// IsAlreadyUpToDate return whether binary is already up to date or not.
func IsAlreadyUpToDate(ver Version) bool {
-	return ver.Current == ver.Latest
+	if ver.Current == ver.Latest {
+		return true
+	}
+
+	return strings.Compare(
+		strings.TrimLeft(ver.Current, "v"),
+		strings.TrimLeft(ver.Latest, "v"),
+	) >= 0
}
```
```go
package main

import (
	"strings"
	"testing"
)

type Version struct {
	Current string // before
	Latest  string // after
}

func IsAlreadyUpToDate(ver Version) bool {
	if ver.Current == ver.Latest {
		return true
	}

	return strings.Compare(
		strings.TrimLeft(ver.Current, "v"),
		strings.TrimLeft(ver.Latest, "v"),
	) >= 0
}

func TestIsAlreadyUpToDate(t *testing.T) {
	for _, test := range []struct {
		curr   string
		latest string
		expect bool
	}{
		// Regular cases
		{curr: "v1.9.0", latest: "v1.9.1", expect: false},
		{curr: "v1.9.0", latest: "v1.9.0", expect: true},
		{curr: "v1.9.1", latest: "v1.9.0", expect: true},

		// Irregular cases
		{curr: "v1.9.1-0.20220908165354-f7355b5d2afa", latest: "v1.9.0", expect: true},
		{
			curr:   "v0.0.0-20220913151710-7c6e287988f3",
			latest: "v0.0.0-20210608161538-9736a4bde949",
			expect: true,
		},
		{
			curr:   "v0.0.0-20210608161538-9736a4bde949",
			latest: "v0.0.0-20220913151710-7c6e287988f3",
			expect: false,
		},
		{curr: "v1.9.0", latest: "1.9.1", expect: false},
		{curr: "1.9.0", latest: "v1.9.1", expect: false},
	} {
		verTmp := Version{
			Current: test.curr,
			Latest:  test.latest,
		}

		expect := test.expect
		actual := IsAlreadyUpToDate(verTmp)

		// Assert
		if expect != actual {
			t.Errorf("(\"%v\" >= \"%v\") = %v, want %v",
				test.curr, test.latest, actual, expect)
		}
	}
}
```

- [View online](https://go.dev/play/p/ZXaGtCCzt8K) @ Go Playground

**Comment by nao1215** [comment]
@KEINOS 
Thanks for the detailed bug report. I have read the sample code you have written.
I think the modification method should be the same as the sample code.

I will find time to fix it.
（However, I may not be able to fix it faster than you expect）

**Comment by KEINOS** [comment]
@nao1215 

> I think the modification method should be the same as the sample code.

OK! If you don't mind, I'd like to do some PR to help solve this issue.

**Comment by nao1215** [comment]
@KEINOS 
My son has gone to bed. I have time to write some code.
However, you may be able to give me a PR. Therefore, today I am expanding the test code.

**Comment by KEINOS** [comment]
OK. Then, I will PR the fix without the test!

**Comment by KEINOS** [comment]
@nao1215 

I did PR the fix @ #39.

By the way, I think I can help to cover up coverages for `github.com/nao1215/gup/internal/goutil`.
I will draft PR as soon as your #38 gets merged.

**Comment by nao1215** [comment]
@KEINOS 

I merged PR #39.

> By the way, I think I can help to cover up coverages for github.com/nao1215/gup/internal/goutil.
I will draft PR as soon as your https://github.com/nao1215/gup/pull/38 gets merged.

٩(｡˃ ᵕ ˂ )و　WRYYYYYYYYYYYYYYYYYYYYYYYYYY
Very nice suggestion!!  I will merge #38 once.
